qiskit_nature/problems/sampling/protein_folding/residue_validator.py

```diff
@@ -0,0 +1,68 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Validates protein residues provided."""
+from typing import List
+
+from .exceptions.invalid_residue_exception import (
+    InvalidResidueException,
+)
+
+
+def _validate_residue_sequence(residue_sequence: List[str]):
+    """
+    Checks if the provided residue sequence contains allowed characters.
+    Args:
+        residue_sequence: A list that contains characters defining residues for a chain of proteins.
+
+    Throws:
+        InvalidResidueException: If an illegal residue character is discovered.
+    """
+    for residue_symbol in residue_sequence:
+        _validate_residue_symbol(residue_symbol)
+
+
+def _validate_residue_symbol(residue_symbol: str):
+    """
+    Checks if the provided residue character is legal. If not, an InvalidResidueException is thrown.
+    Args:
+        residue_symbol: symbol of a residue.
+
+    Raises:
+        InvalidResidueException: if a symbol provided is not legal.
+    """
+    valid_residues = [
```

Comment:
Users use symbols all the time whenever they define a new problem. Each chain needs to have symbols specified for each bead. Not sure if the lookup is the way to go, though... @pbark ?

Reply:
A user should specify proteins with these symbols. Each symbol corresponds to an amminoacid as part of the sequence. I think that what @woodsp-ibm is proposing is that if one would like to print the symbol he can get the name of the amminoacid. I  think he can do it, although this will not really extend our functionality and further.

Reply:
I guess these symbols are well known and recognized easy enough by people - it was not really about extending the function more around end-user usability that if you printed things you could get the full name for any symbol. Maybe for the targetted  community the symbols are usable enough?

Reply:
Yes, these symbols are well known in the target community. I remember asking for the comments with the full names in my original review purely thinking of future developers looking at this code (which may not be familiar with these symbols).

One thing that comes to mind now is the order of the symbols? Does that have meaning? If yes, we should at least comment on that (also to prevent a future developer from reordering the list (e.g. alphabetically))

Reply:
I believe the order has no meaning.

Reply:
If none of the indices for the entries are used, I suggest we put them into some meaningful order: e.g. alphabetically.